File: src/publishstr.py
#!/usr/bin/env python3
#chmod u+x ./beginner_tutorials/src/publishstr.py
import rospy
from std_msgs.msg import String

rospy.init_node('string_pub')
pub = rospy.Publisher('/words', String, queue_size=10)

msg = String("slm knk")
rate = rospy.Rate(10)

while not rospy.is_shutdown():
    print(f"Published: {msg}")
    pub.publish(msg)
    rate.sleep()
    

File: src/publishStr.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

using namespace std;

int main(int argc, char **argv)
{
    ros::init(argc, argv, "stringpublisher");
    ros::NodeHandle n;
    ros::Publisher word_pub = n.advertise<std_msgs::String>("/words", 10);
    ros::Rate loop_rate(10);
    int count = 0;
    while (ros::ok())
    {
        std_msgs::String msg;
        stringstream ss;
        ss << "Hi sup' " << count;
        count++;

        msg.data = ss.str();

        cout << msg.data << endl;

        word_pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;
}
